src/main/java/seedu/address/model/person/predicates/StaffHasCorrectIndexPredicate.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.person.predicates;
+
+import java.util.function.Predicate;
+
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Index} matches the index given.
+ *
+ * This class is used in FindCommand.
+ */
+public class StaffHasCorrectIndexPredicate implements Predicate<Person> {
+    private final int index;
+    private final Model model;
+    private final Person correctPerson;
+
+    /**
+     * Constructs a StaffHasCorrectIndexPredicate object which returns true if the person tested against is the same as
+     * the person corresponding to the index inputted.
+     *
+     * @param index The index that the user inputted.
+     * @param model The model containing the list of Persons who will be searched from.
+     */
+    public StaffHasCorrectIndexPredicate(int index, Model model) {
+        this.index = index;
+        this.model = model;
+        this.correctPerson = model.getFilteredPersonListByIndex(index);
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return correctPerson.isSamePerson(person);
+    }
+
+    /**
+     * Returns the index of a StaffHasCorrectIndexPredicate object.
+     *
+     * @return the index of a StaffHasCorrectIndexPredicate object.
+     */
+    public int getIndex() {
```

Comment:
There is no need for this method? Only use i have seen is in the equals method and the index can be directly accessed.

Reply:
Does it not break some kinda abstraction to access another object's attributes directly? I might be wrong about this though